internal/brokerstore/brokerstore.go

```diff
@@ -69,6 +69,9 @@ type IBrokerStore interface {
 	// GetProducer returns for an existing producer instance, if available returns that else creates as new instance
 	GetProducer(ctx context.Context, op messagebroker.ProducerClientOptions) (messagebroker.Producer, error)
 
+	// ShutdownAndRemoveProducer deletes the producer from the store followed by a shutdown
+	ShutdownAndRemoveProducer(ctx context.Context, op messagebroker.ProducerClientOptions) bool
```

Comment:
similarly to keep it consistent with consumer interfaces

GetProducer
RemoveProducer